packages/flutter/test/material/flat_button_test.dart

```diff
@@ -333,6 +333,79 @@ void main() {
         paintsExactlyCountTimes(#clipPath, 0),
     );
   });
+
+  testWidgets('FlatButton responds to tap and onLongPress when enabled', (WidgetTester tester) async {
+
+    int pressedCount = 0;
+
+    Widget buildFrame({VoidCallback onPressed, VoidCallback onLongPress}) {
+      return Directionality(
+        textDirection: TextDirection.ltr,
+        child: FlatButton(child: const Text('button'), onPressed: onPressed, onLongPress: onLongPress),
+      );
+    }
+
+    // onPressed not null, onLongPress null.
+    await tester.pumpWidget(
+      buildFrame(onPressed: () { pressedCount += 1; }, onLongPress: null),
+    );
+    expect(tester.widget<FlatButton>(find.byType(FlatButton)).enabled, true);
```

Comment:
This is another example where it might be difficult to understand what is being tested off the bat. Maybe something like: 
```suggestion
    FlatButton flatButton = tester.widget<FlatButton>(find.byType(FlatButton));
    expect(flatButton.enabled, true);
```

Or, what you have in your other test